### PR TITLE
screen docs

### PR DESCRIPTION
**Pull request by willmcgugan**
Updating screen docs to document `push_screen` with `wait_for_dismiss`. However in doing so, I realised that I should add `push_screen_wait` for brevity.

Made a tweak to toasts, but giving them a slightly thicker line.

### CODE REVIEW

File: docs/guide/screens.md [hunk 1]
```diff
@@ -258,6 +258,45 @@ You may have noticed in the previous example that we changed the base class to `
 The addition of `[bool]` adds typing information that tells the type checker to expect a boolean in the call to `dismiss`, and that any callback set in `push_screen` should also expect the same type. As always, typing is optional in Textual, but this may help you catch bugs.
 
 
+### Waiting for screens
+
+It is also possible to wait on a screen to be dismissed, which can feel like a more natural way of expressing logic that a callback.
```
Comment (rodrigogiraoserrao) [inline docs/guide/screens.md:263]:
```suggestion
It is also possible to wait on a screen to be dismissed, which can feel like a more natural way of expressing logic than a callback.
```

File: docs/guide/screens.md [hunk 2]
```diff
@@ -258,6 +258,45 @@ You may have noticed in the previous example that we changed the base class to `
...
+The [`push_screen_wait()`][textual.app.App.push_screen_wait] method will push a screen and wait for the its result (value from [`Screen.dismiss()`][textual.screen.Screen.dismiss]).
```
Comment (rodrigogiraoserrao) [inline docs/guide/screens.md:264]:
```suggestion
The [`push_screen_wait()`][textual.app.App.push_screen_wait] method will push a screen and wait for its result (the value from [`Screen.dismiss()`][textual.screen.Screen.dismiss]).
```

File: src/textual/app.py [hunk 3]
```diff
@@ -1945,6 +1945,29 @@ def push_screen(
...
+    ) -> ScreenResultType:
+        ...
+
+    @overload
+    async def push_screen_wait(self, screen: str) -> Any:
+        ...
+
+    async def push_screen_wait(
+        self, screen: Screen[ScreenResultType] | str
+    ) -> ScreenResultType:
```
Comment (rodrigogiraoserrao) [inline src/textual/app.py:1960]: I don't understand why `-> ScreenResultType | Any` isn't needed.

Comment (willmcgugan) [reply src/textual/app.py:1960]: I think you're right. I don't know why it didn't complain.

File: docs/guide/screens.md [hunk 4]
```diff
@@ -258,6 +258,45 @@ You may have noticed in the previous example that we changed the base class to `
...
+    ```css title="questions01.tcss"
+    --8<-- "docs/examples/guide/screens/questions01.tcss"
+    ```
+
+=== "Output"
+
+    ```{.textual path="docs/examples/guide/screens/questions01.py"}
+    ```
+
+The mount handler on the app is decorated with `@work`, which makes the code run asynchronously.
```
[the remark this answers is not in the document]
Comment (willmcgugan) [reply docs/guide/screens.md:294]: It's in a worker so it doesn't prevent more messages being processed.

I'll reword it to avoid confusion.
